pkg/controller/importconfig/render.go

```diff
@@ -31,6 +37,137 @@ type ImagePullSecretConfig struct {
 	ImagePullSecretType      corev1.SecretType
 }
 
-const (
-	AgentRegistrationDefaultBootstrapSAName = "agent-registration-bootstrap"
-)
+type KlusterletManifestsConfig struct {
+	Files       []string
+	InstallMode operatorv1.InstallMode
+
+	ClusterName                  string
+	KlusterletNamespace          string
+	KlusterletClusterAnnotations map[string]string
+	BootstrapKubeconfig          []byte
+
+	// In the managed cluster annotations, it contains nodeSelectors and tolerations for the klusterlet deployment.
+	ManagedClusterAnnotations map[string]string
+
+	generateImagePullSecret bool // by default is true, in hosted mode, it will be set false
+}
+
+func NewKlusterletManifestsConfig(files []string, installMode operatorv1.InstallMode,
```

Comment:
why files is an input? it seems to me the file is differently only when mode is different, can we use mode to know which files should be added?

Reply:
Fixed.